src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -765,78 +766,139 @@ function saturation_adjustment(e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
   end
 end
 
-
 """
-    saturation_adjustment_q_tot_θ_liq_ice(θ_liq_ice, q_tot, ρ, p)
+    saturation_adjustment_q_tot_θ_liq_ice(θ_liq_ice, q_tot, ρ)
 
 Compute the temperature that is consistent with
 
  - `θ_liq_ice` liquid-ice potential temperature
  - `q_tot` total specific humidity
  - `ρ` density
- - `p` pressure
 
 See also [`saturation_adjustment`](@ref).
 """
-function saturation_adjustment_q_tot_θ_liq_ice(θ_liq_ice::FT, q_tot::FT, ρ::FT, p::FT) where {FT<:Real}
-  T_1 = air_temperature_from_liquid_ice_pottemp(θ_liq_ice, p) # Assume all vapor
+function saturation_adjustment_q_tot_θ_liq_ice(θ_liq_ice::FT, q_tot::FT, ρ::FT) where {FT<:Real}
+  # TODO/FIXME: The initial guesses are computed via iteration,
+  # which is potentially expensive. Other options should
+  # be investigated. Also, the tolerance/iterations are large
+  # when iterated. Perhaps this can be reduced with better guesses.
+
+  # TODO/FIXME: The iteration process seems brittle, and needs to
+  # be reviewed/improved
+  T_1 = air_temperature_initial_guess(θ_liq_ice, ρ, PhasePartition(q_tot)) # Assume all vapor
   q_v_sat = q_vap_saturation(T_1, ρ)
-  if q_tot <= q_v_sat # If not saturated
+  saturated = q_tot <= q_v_sat
+  if saturated
     return T_1
-  else  # If saturated, iterate
-    T_2 = air_temperature_from_liquid_ice_pottemp(θ_liq_ice, p, PhasePartition(q_tot, FT(0), q_tot)) # Assume all ice
+  else
+    T_2 = air_temperature_initial_guess(θ_liq_ice, ρ, PhasePartition(q_tot, FT(0), q_tot)) # Assume all vapor
```

Comment:
It seems very odd to get an initial guess by secant iterations, before doing more secant iterations. Why not trying the all vapor initial guess, which in most cases will be quite good:
```suggestion
    T_2 = air_temperature_from_liquid_ice_pottemp(θ_liq_ice, ρ, PhasePartition(q_tot, FT(0), FT(0)))
```